engine/src/main/java/org/hibernate/search/bridge/builtin/time/impl/InstantBridge.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.bridge.builtin.time.impl;
+
+import static java.time.temporal.ChronoField.INSTANT_SECONDS;
+import static java.time.temporal.ChronoField.NANO_OF_SECOND;
+
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.format.SignStyle;
+
+/**
+ * Converts a {@link Instant} to a {@link String}.
+ * <p>
+ * The string is obtained concatenating the number of seconds from Epoch with the nano of seconds.
+ * The values are padded with 0 to allow field sorting.
+ *
+ * @author Davide D'Alto
+ */
+public class InstantBridge extends TemporalAccessorStringBridge<Instant> {
```

Comment:
The same for instants; Storing it as a numeric field seems more appropriate to me.